install.sh is failing to build libffi on gentoo

```
techno@rawbit joinmarket-clientserver % ./install.sh 

        INFO: Joinmarket-Qt for GUI Taker and Tumbler modes is available.
        Install Qt dependencies (~160mb)? (y/N) N
Not building Qt GUI
OS can not be determined. Trying to build.
Directory jmvenv exists.  Remove and recreate? (y/N) y
Requirement already satisfied: pip in ./jmvenv/lib/python3.12/site-packages (24.3.1)
Collecting setuptools
  Using cached setuptools-75.6.0-py3-none-any.whl.metadata (6.7 kB)
Using cached setuptools-75.6.0-py3-none-any.whl (1.2 MB)
Installing collected packages: setuptools
Successfully installed setuptools-75.6.0
~/code/joinmarket-clientserver/deps ~/code/joinmarket-clientserver
~/code/joinmarket-clientserver/deps/cache ~/code/joinmarket-clientserver/deps ~/code/joinmarket-clientserver
v0.5.0.tar.gz: OK
v0.5.0.tar.gz: OK
~/code/joinmarket-clientserver/deps ~/code/joinmarket-clientserver
~/code/joinmarket-clientserver/deps/secp256k1-0.5.0 ~/code/joinmarket-clientserver/deps ~/code/joinmarket-clientserver
test -z "precompute_ecmult precompute_ecmult_gen" || rm -f precompute_ecmult precompute_ecmult_gen
test -z "libsecp256k1.la" || rm -f libsecp256k1.la
rm -rf .libs _libs
test -z "libsecp256k1_precomputed.la " || rm -f libsecp256k1_precomputed.la 
rm -f *.o
test -z "noverify_tests.log tests.log exhaustive_tests.log" || rm -f noverify_tests.log tests.log exhaustive_tests.log
rm -f *.lo
rm -rf src/.libs src/_libs
 rm -f noverify_tests tests exhaustive_tests
rm -rf src/asm/.libs src/asm/_libs
rm -f examples/*.o
test -z "noverify_tests.trs tests.trs exhaustive_tests.trs" || rm -f noverify_tests.trs tests.trs exhaustive_tests.trs
rm -f src/*.o
test -z "test-suite.log" || rm -f test-suite.log
rm -f ./so_locations
rm -f ./so_locations
rm -f src/*.lo
rm -f src/asm/*.o
rm -f src/asm/*.lo
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'build-aux'.
libtoolize: copying file 'build-aux/ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'build-aux/m4'.
libtoolize: copying file 'build-aux/m4/libtool.m4'
libtoolize: copying file 'build-aux/m4/ltoptions.m4'
libtoolize: copying file 'build-aux/m4/ltsugar.m4'
libtoolize: copying file 'build-aux/m4/ltversion.m4'
libtoolize: copying file 'build-aux/m4/lt~obsolete.m4'
configure.ac:37: installing 'build-aux/compile'
configure.ac:27: installing 'build-aux/missing'
Makefile.am: installing 'build-aux/depcomp'
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether gmake sets $(MAKE)... yes
checking whether gmake supports nested variables... yes
checking whether gmake supports nested variables... (cached) yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking whether gcc understands -c and -o together... yes
checking whether gmake supports the include directive... yes (GNU style)
checking dependency style of gcc... gcc3
checking dependency style of gcc... gcc3
checking for ar... ar
checking the archiver (ar) interface... ar
checking how to print strings... printf
checking for a sed that does not truncate output... /bin/sed
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for fgrep... /bin/grep -F
checking for ld used by gcc... /usr/x86_64-pc-linux-gnu/bin/ld
checking if the linker (/usr/x86_64-pc-linux-gnu/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/x86_64-pc-linux-gnu/bin/ld option to reload object files... -r
checking for file... file
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... dlltool
checking how to associate runtime and link libraries... printf %s\n
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for a working dd... /bin/dd
checking how to truncate binary pipes... /bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... yes
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/usr/x86_64-pc-linux-gnu/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking if gcc supports -Werror... yes
checking if gcc supports -std=c89 -pedantic -Wno-long-long -Wnested-externs -Wshadow -Wstrict-prototypes -Wundef... yes
checking if gcc supports -Wno-overlength-strings... yes
checking if gcc supports -Wall... yes
checking if gcc supports -Wno-unused-function... yes
checking if gcc supports -Wextra... yes
checking if gcc supports -Wcast-align... yes
checking if gcc supports -Wcast-align=strict... yes
checking if gcc supports -Wconditional-uninitialized... no
checking if gcc supports -Wreserved-identifier... no
checking if gcc supports -fvisibility=hidden... yes
checking for valgrind support... 
checking for x86_64 assembly availability... yes
configure: ******
configure: WARNING: experimental build
configure: Experimental features do not have stable APIs or properties, and may not be safe for production use.
configure: ******
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating libsecp256k1.pc
config.status: executing depfiles commands
config.status: executing libtool commands

Build Options:
  with external callbacks = no
  with benchmarks         = no
  with tests              = yes
  with ctime tests        = no
  with coverage           = no
  with examples           = no
  module ecdh             = yes
  module recovery         = yes
  module extrakeys        = yes
  module schnorrsig       = yes
  module ellswift         = yes

  asm                     = x86_64
  ecmult window size      = 15
  ecmult gen table size   = 22 KiB

  valgrind                = no
  CC                      = gcc
  CPPFLAGS                = 
  SECP_CFLAGS             = -O2  -std=c89 -pedantic -Wno-long-long -Wnested-externs -Wshadow -Wstrict-prototypes -Wundef -Wno-overlength-strings -Wall -Wno-unused-function -Wextra -Wcast-align -Wcast-align=strict -fvisibility=hidden 
  CFLAGS                  = -g -O2
  LDFLAGS                 = 
gmake  all-am
gmake[1]: Entering directory '/home/techno/code/joinmarket-clientserver/deps/secp256k1-0.5.0'
  CC       src/noverify_tests-tests.o
  CC       src/libsecp256k1_precomputed_la-precomputed_ecmult.lo
  CC       src/libsecp256k1_precomputed_la-precomputed_ecmult_gen.lo
  CC       src/tests-tests.o
  CC       src/exhaustive_tests-tests_exhaustive.o
  CC       src/libsecp256k1_la-secp256k1.lo
  CCLD     libsecp256k1_precomputed.la
  CCLD     exhaustive_tests
  CCLD     libsecp256k1.la
  CCLD     noverify_tests
  CCLD     tests
gmake[1]: Leaving directory '/home/techno/code/joinmarket-clientserver/deps/secp256k1-0.5.0'
gmake  check-am
gmake[1]: Entering directory '/home/techno/code/joinmarket-clientserver/deps/secp256k1-0.5.0'
gmake  check-TESTS
gmake[2]: Entering directory '/home/techno/code/joinmarket-clientserver/deps/secp256k1-0.5.0'
gmake[3]: Entering directory '/home/techno/code/joinmarket-clientserver/deps/secp256k1-0.5.0'
PASS: exhaustive_tests
PASS: noverify_tests
PASS: tests
============================================================================
Testsuite summary for libsecp256k1 0.5.0
============================================================================
# TOTAL: 3
# PASS:  3
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
gmake[3]: Leaving directory '/home/techno/code/joinmarket-clientserver/deps/secp256k1-0.5.0'
gmake[2]: Leaving directory '/home/techno/code/joinmarket-clientserver/deps/secp256k1-0.5.0'
gmake[1]: Leaving directory '/home/techno/code/joinmarket-clientserver/deps/secp256k1-0.5.0'
gmake  install-am
gmake[1]: Entering directory '/home/techno/code/joinmarket-clientserver/deps/secp256k1-0.5.0'
gmake[2]: Entering directory '/home/techno/code/joinmarket-clientserver/deps/secp256k1-0.5.0'
 /bin/mkdir -p '/home/techno/code/joinmarket-clientserver/jmvenv/lib'
 /bin/mkdir -p '/home/techno/code/joinmarket-clientserver/jmvenv/include'
 /bin/mkdir -p '/home/techno/code/joinmarket-clientserver/jmvenv/lib/pkgconfig'
 /bin/sh ./libtool   --mode=install /usr/bin/install -c   libsecp256k1.la '/home/techno/code/joinmarket-clientserver/jmvenv/lib'
 /usr/bin/install -c -m 644 include/secp256k1.h include/secp256k1_preallocated.h include/secp256k1_ecdh.h include/secp256k1_recovery.h include/secp256k1_extrakeys.h include/secp256k1_schnorrsig.h include/secp256k1_ellswift.h '/home/techno/code/joinmarket-clientserver/jmvenv/include'
 /usr/bin/install -c -m 644 libsecp256k1.pc '/home/techno/code/joinmarket-clientserver/jmvenv/lib/pkgconfig'
libtool: install: /usr/bin/install -c .libs/libsecp256k1.so.2.2.0 /home/techno/code/joinmarket-clientserver/jmvenv/lib/libsecp256k1.so.2.2.0
libtool: install: (cd /home/techno/code/joinmarket-clientserver/jmvenv/lib && { ln -s -f libsecp256k1.so.2.2.0 libsecp256k1.so.2 || { rm -f libsecp256k1.so.2 && ln -s libsecp256k1.so.2.2.0 libsecp256k1.so.2; }; })
libtool: install: (cd /home/techno/code/joinmarket-clientserver/jmvenv/lib && { ln -s -f libsecp256k1.so.2.2.0 libsecp256k1.so || { rm -f libsecp256k1.so && ln -s libsecp256k1.so.2.2.0 libsecp256k1.so; }; })
libtool: install: /usr/bin/install -c .libs/libsecp256k1.lai /home/techno/code/joinmarket-clientserver/jmvenv/lib/libsecp256k1.la
libtool: install: /usr/bin/install -c .libs/libsecp256k1.a /home/techno/code/joinmarket-clientserver/jmvenv/lib/libsecp256k1.a
libtool: install: chmod 644 /home/techno/code/joinmarket-clientserver/jmvenv/lib/libsecp256k1.a
libtool: install: ranlib /home/techno/code/joinmarket-clientserver/jmvenv/lib/libsecp256k1.a
libtool: finish: PATH="/home/techno/code/joinmarket-clientserver/jmvenv/bin:/home/techno/.poetry/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/bin:/usr/lib/llvm/19/bin:/usr/lib/llvm/18/bin:/sbin" ldconfig -n /home/techno/code/joinmarket-clientserver/jmvenv/lib
----------------------------------------------------------------------
Libraries have been installed in:
   /home/techno/code/joinmarket-clientserver/jmvenv/lib

If you ever happen to want to link against installed libraries
in a given directory, LIBDIR, you must either use libtool, and
specify the full pathname of the library, or use the '-LLIBDIR'
flag during linking and do at least one of the following:
   - add LIBDIR to the 'LD_LIBRARY_PATH' environment variable
     during execution
   - add LIBDIR to the 'LD_RUN_PATH' environment variable
     during linking
   - use the '-Wl,-rpath -Wl,LIBDIR' linker flag
   - have your system administrator add LIBDIR to '/etc/ld.so.conf'

See any operating system documentation about shared libraries for
more information, such as the ld(1) and ld.so(8) manual pages.
----------------------------------------------------------------------
gmake[2]: Leaving directory '/home/techno/code/joinmarket-clientserver/deps/secp256k1-0.5.0'
gmake[1]: Leaving directory '/home/techno/code/joinmarket-clientserver/deps/secp256k1-0.5.0'
~/code/joinmarket-clientserver/deps ~/code/joinmarket-clientserver
~/code/joinmarket-clientserver/deps/cache ~/code/joinmarket-clientserver/deps ~/code/joinmarket-clientserver
v3.2.1.tar.gz: OK
v3.2.1.tar.gz: OK
~/code/joinmarket-clientserver/deps ~/code/joinmarket-clientserver
~/code/joinmarket-clientserver/deps/libffi-3.2.1 ~/code/joinmarket-clientserver/deps ~/code/joinmarket-clientserver
patching file Makefile.am
patching file configure.ac
patching file configure.ac
autoreconf-2.72: export WARNINGS=
autoreconf-2.72: Entering directory '.'
autoreconf-2.72: configure.ac: not using Gettext
autoreconf-2.72: running: aclocal --force -I m4
autoreconf-2.72: configure.ac: tracing
autoreconf-2.72: running: libtoolize --copy --force
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, '.'.
libtoolize: copying file './ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
libtoolize: copying file 'm4/libtool.m4'
libtoolize: copying file 'm4/ltoptions.m4'
libtoolize: copying file 'm4/ltsugar.m4'
libtoolize: copying file 'm4/ltversion.m4'
libtoolize: copying file 'm4/lt~obsolete.m4'
autoreconf-2.72: configure.ac: not using Intltool
autoreconf-2.72: configure.ac: not using Gtkdoc
autoreconf-2.72: running: aclocal --force -I m4
autoreconf-2.72: running: /usr/bin/autoconf-2.72 --force
configure.ac:8: warning: The macro 'AC_CANONICAL_SYSTEM' is obsolete.
configure.ac:8: You should run autoupdate.
./lib/autoconf/general.m4:2081: AC_CANONICAL_SYSTEM is expanded from...
configure.ac:8: the top level
configure.ac:42: warning: The macro 'AC_PROG_LIBTOOL' is obsolete.
configure.ac:42: You should run autoupdate.
m4/libtool.m4:100: AC_PROG_LIBTOOL is expanded from...
configure.ac:42: the top level
configure.ac:69: warning: The macro 'AC_TRY_COMPILE' is obsolete.
configure.ac:69: You should run autoupdate.
./lib/autoconf/general.m4:2845: AC_TRY_COMPILE is expanded from...
lib/m4sugar/m4sh.m4:690: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:697: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
acinclude.m4:3: AC_FUNC_MMAP_BLACKLIST is expanded from...
configure.ac:69: the top level
configure.ac:342: warning: The macro 'AC_HEADER_STDC' is obsolete.
configure.ac:342: You should run autoupdate.
./lib/autoconf/headers.m4:663: AC_HEADER_STDC is expanded from...
configure.ac:342: the top level
configure.ac:369: warning: The macro 'AC_TRY_COMPILE' is obsolete.
configure.ac:369: You should run autoupdate.
./lib/autoconf/general.m4:2845: AC_TRY_COMPILE is expanded from...
lib/m4sugar/m4sh.m4:690: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:697: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
m4/asmcfi.m4:1: GCC_AS_CFI_PSEUDO_OP is expanded from...
configure.ac:369: the top level
configure.ac:372: warning: The macro 'AC_TRY_LINK' is obsolete.
configure.ac:372: You should run autoupdate.
./lib/autoconf/general.m4:2918: AC_TRY_LINK is expanded from...
lib/m4sugar/m4sh.m4:690: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:697: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
configure.ac:372: the top level
configure.ac:388: warning: The macro 'AC_TRY_COMPILE' is obsolete.
configure.ac:388: You should run autoupdate.
./lib/autoconf/general.m4:2845: AC_TRY_COMPILE is expanded from...
lib/m4sugar/m4sh.m4:690: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:697: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
configure.ac:388: the top level
configure.ac:416: warning: The macro 'AC_TRY_COMPILE' is obsolete.
configure.ac:416: You should run autoupdate.
./lib/autoconf/general.m4:2845: AC_TRY_COMPILE is expanded from...
lib/m4sugar/m4sh.m4:690: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:697: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
configure.ac:416: the top level
configure.ac:429: warning: The macro 'AC_TRY_COMPILE' is obsolete.
configure.ac:429: You should run autoupdate.
./lib/autoconf/general.m4:2845: AC_TRY_COMPILE is expanded from...
lib/m4sugar/m4sh.m4:690: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:697: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
configure.ac:429: the top level
configure.ac:547: warning: The macro 'AC_HELP_STRING' is obsolete.
configure.ac:547: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
configure.ac:547: the top level
autoreconf-2.72: running: /usr/bin/autoheader-2.72 --force
autoreconf-2.72: running: automake --add-missing --copy --force-missing
configure.ac:32: installing './compile'
configure.ac:20: installing './missing'
Makefile.am: installing './depcomp'
autoreconf-2.72: './install-sh' is updated
autoreconf-2.72: Leaving directory '.'
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking target system type... x86_64-pc-linux-gnu
continue configure in default builddir "./x86_64-pc-linux-gnu"
....exec /bin/sh .././configure "--srcdir=.." "--enable-builddir=x86_64-pc-linux-gnu" "linux
gnu"
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking target system type... x86_64-pc-linux-gnu
checking for gsed... /bin/gsed
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking whether gcc understands -c and -o together... yes
checking whether make supports the include directive... yes (GNU style)
checking dependency style of gcc... gcc3
checking for g++... g++
checking whether the compiler supports GNU C++... yes
checking whether g++ accepts -g... yes
checking for g++ option to enable C++11 features... none needed
checking dependency style of g++... gcc3
checking dependency style of gcc... gcc3
checking how to print strings... printf
checking for a sed that does not truncate output... /bin/sed
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for fgrep... /bin/grep -F
checking for ld used by gcc... /usr/x86_64-pc-linux-gnu/bin/ld
checking if the linker (/usr/x86_64-pc-linux-gnu/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/x86_64-pc-linux-gnu/bin/ld option to reload object files... -r
checking for file... file
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... dlltool
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for a working dd... /bin/dd
checking how to truncate binary pipes... /bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... yes
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/usr/x86_64-pc-linux-gnu/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking how to run the C++ preprocessor... g++ -E
checking for ld used by g++... /usr/x86_64-pc-linux-gnu/bin/ld -m elf_x86_64
checking if the linker (/usr/x86_64-pc-linux-gnu/bin/ld -m elf_x86_64) is GNU ld... yes
checking whether the g++ linker (/usr/x86_64-pc-linux-gnu/bin/ld -m elf_x86_64) supports shared libraries... yes
checking for g++ option to produce PIC... -fPIC -DPIC
checking if g++ PIC flag -fPIC -DPIC works... yes
checking if g++ static flag -static works... yes
checking if g++ supports -c -o file.o... yes
checking if g++ supports -c -o file.o... (cached) yes
checking whether the g++ linker (/usr/x86_64-pc-linux-gnu/bin/ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking size of size_t... 8
checking for C compiler vendor... gnu
checking whether C compiler accepts -fstrict-aliasing... yes
checking whether C compiler accepts -ffast-math... yes
checking for gcc architecture flag... 
checking for x86 cpuid 0 output... 16:756e6547:6c65746e:49656e69
checking for x86 cpuid 1 output... 906ea:b100800:7ffafbff:bfebfbff
checking whether C compiler accepts -march=core2... yes
checking for gcc architecture flag... -march=core2
checking whether C compiler accepts -O3 -fomit-frame-pointer -fstrict-aliasing -ffast-math -march=core2... yes
checking CFLAGS for maximum warnings... -Wall
checking whether to enable maintainer-specific portions of Makefiles... no
checking for sys/mman.h... yes
checking for mmap... yes
checking for mkostemp... yes
checking for sys/mman.h... (cached) yes
checking for mmap... (cached) yes
checking whether read-only mmap of a plain file works... yes
checking whether mmap from /dev/zero works... yes
checking for MAP_ANON(YMOUS)... yes
checking whether mmap with MAP_ANON(YMOUS) works... yes
checking for egrep... (cached) /bin/grep -E
checking for memcpy... yes
checking for size_t... yes
checking for working alloca.h... yes
checking for alloca... yes
checking size of double... 8
checking size of long double... 16
checking whether byte ordering is bigendian... no
checking assembler .cfi pseudo-op support... yes
checking assembler supports pc related relocs... yes
checking assembler .ascii pseudo-op support... yes
checking assembler .string pseudo-op support... yes
checking for _ prefix in compiled symbols... no
checking toolchain supports unwind section type... yes
checking whether .eh_frame section should be read-only... yes
checking for __attribute__((visibility("hidden")))... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating include/Makefile
config.status: creating include/ffi.h
config.status: creating Makefile
config.status: creating testsuite/Makefile
config.status: creating man/Makefile
config.status: creating libffi.pc
config.status: creating fficonfig.h
config.status: fficonfig.h is unchanged
config.status: linking ../src/x86/ffitarget.h to include/ffitarget.h
config.status: executing buildir commands
config.status: create top_srcdir/Makefile guessed from local Makefile
config.status: build in x86_64-pc-linux-gnu (HOST=)
config.status: executing depfiles commands
config.status: executing libtool commands
config.status: executing include commands
config.status: executing src commands
MAKE x86_64-pc-linux-gnu : 0 * uninstall
gmake[1]: Entering directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu'
Making uninstall in include
gmake[2]: Entering directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu/include'
gmake[2]: Leaving directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu/include'
Making uninstall in testsuite
gmake[2]: Entering directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu/testsuite'
gmake[2]: Nothing to be done for 'uninstall'.
gmake[2]: Leaving directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu/testsuite'
Making uninstall in man
gmake[2]: Entering directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu/man'
gmake[2]: Leaving directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu/man'
gmake[2]: Entering directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu'
 ( cd '/home/techno/code/joinmarket-clientserver/jmvenv/lib/pkgconfig' && rm -f libffi.pc )
 /bin/sh ./libtool   --mode=uninstall rm -f '/home/techno/code/joinmarket-clientserver/jmvenv/lib/../lib64/libffi.la'
gmake[2]: Leaving directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu'
gmake[1]: Leaving directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu'
MAKE x86_64-pc-linux-gnu : 0 * all-all
gmake[1]: Entering directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu'
gmake 'AR_FLAGS=' 'CC_FOR_BUILD=' 'CFLAGS=-O3 -fomit-frame-pointer -fstrict-aliasing -ffast-math -march=core2  -Wall -fexceptions' 'CXXFLAGS=-g -O2' 'CFLAGS_FOR_BUILD=' 'CFLAGS_FOR_TARGET=' 'INSTALL=/usr/bin/install -c' 'INSTALL_DATA=/usr/bin/install -c -m 644' 'INSTALL_PROGRAM=/usr/bin/install -c' 'INSTALL_SCRIPT=/usr/bin/install -c' 'JC1FLAGS=' 'LDFLAGS=' 'LIBCFLAGS=' 'LIBCFLAGS_FOR_TARGET=' 'MAKE=gmake' 'MAKEINFO=/bin/sh '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/missing' makeinfo ' 'PICFLAG=' 'PICFLAG_FOR_TARGET=' 'RUNTESTFLAGS=' 'SHELL=/bin/sh' 'exec_prefix=/home/techno/code/joinmarket-clientserver/jmvenv' 'infodir=/home/techno/code/joinmarket-clientserver/jmvenv/share/info' 'libdir=/home/techno/code/joinmarket-clientserver/jmvenv/lib' 'mandir=/home/techno/code/joinmarket-clientserver/jmvenv/share/man' 'prefix=/home/techno/code/joinmarket-clientserver/jmvenv' 'AR=ar' 'AS=as' 'CC=gcc' 'CXX=g++' 'LD=/usr/x86_64-pc-linux-gnu/bin/ld -m elf_x86_64' 'NM=/usr/bin/nm -B' 'RANLIB=ranlib' 'DESTDIR=' all-recursive
gmake[2]: Entering directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu'
Making all in include
gmake[3]: Entering directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu/include'
gmake[3]: Nothing to be done for 'all'.
gmake[3]: Leaving directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu/include'
Making all in testsuite
gmake[3]: Entering directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu/testsuite'
gmake[3]: Nothing to be done for 'all'.
gmake[3]: Leaving directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu/testsuite'
Making all in man
gmake[3]: Entering directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu/man'
gmake[3]: Nothing to be done for 'all'.
gmake[3]: Leaving directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu/man'
gmake[3]: Entering directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu'
depbase=`echo src/prep_cif.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/sh ./libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I..  -I. -I../include -Iinclude -I../src   -O3 -fomit-frame-pointer -fstrict-aliasing -ffast-math -march=core2  -Wall -fexceptions -MT src/prep_cif.lo -MD -MP -MF $depbase.Tpo -c -o src/prep_cif.lo ../src/prep_cif.c &&\
mv -f $depbase.Tpo $depbase.Plo
depbase=`echo src/types.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/sh ./libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I..  -I. -I../include -Iinclude -I../src   -O3 -fomit-frame-pointer -fstrict-aliasing -ffast-math -march=core2  -Wall -fexceptions -MT src/types.lo -MD -MP -MF $depbase.Tpo -c -o src/types.lo ../src/types.c &&\
mv -f $depbase.Tpo $depbase.Plo
depbase=`echo src/raw_api.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/sh ./libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I..  -I. -I../include -Iinclude -I../src   -O3 -fomit-frame-pointer -fstrict-aliasing -ffast-math -march=core2  -Wall -fexceptions -MT src/raw_api.lo -MD -MP -MF $depbase.Tpo -c -o src/raw_api.lo ../src/raw_api.c &&\
mv -f $depbase.Tpo $depbase.Plo
depbase=`echo src/java_raw_api.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/sh ./libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I..  -I. -I../include -Iinclude -I../src   -O3 -fomit-frame-pointer -fstrict-aliasing -ffast-math -march=core2  -Wall -fexceptions -MT src/java_raw_api.lo -MD -MP -MF $depbase.Tpo -c -o src/java_raw_api.lo ../src/java_raw_api.c &&\
mv -f $depbase.Tpo $depbase.Plo
depbase=`echo src/closures.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/sh ./libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I..  -I. -I../include -Iinclude -I../src   -O3 -fomit-frame-pointer -fstrict-aliasing -ffast-math -march=core2  -Wall -fexceptions -MT src/closures.lo -MD -MP -MF $depbase.Tpo -c -o src/closures.lo ../src/closures.c &&\
mv -f $depbase.Tpo $depbase.Plo
depbase=`echo src/x86/ffi64.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/sh ./libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I..  -I. -I../include -Iinclude -I../src   -O3 -fomit-frame-pointer -fstrict-aliasing -ffast-math -march=core2  -Wall -fexceptions -MT src/x86/ffi64.lo -MD -MP -MF $depbase.Tpo -c -o src/x86/ffi64.lo ../src/x86/ffi64.c &&\
mv -f $depbase.Tpo $depbase.Plo
depbase=`echo src/x86/unix64.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/sh ./libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I..  -I. -I../include -Iinclude -I../src  -I. -I../include -Iinclude -I../src  -MT src/x86/unix64.lo -MD -MP -MF $depbase.Tpo -c -o src/x86/unix64.lo ../src/x86/unix64.S &&\
mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I. -I../include -Iinclude -I../src -O3 -fomit-frame-pointer -fstrict-aliasing -ffast-math -march=core2 -Wall -fexceptions -MT src/java_raw_api.lo -MD -MP -MF src/.deps/java_raw_api.Tpo -c ../src/java_raw_api.c  -fPIC -DPIC -o src/.libs/java_raw_api.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I. -I../include -Iinclude -I../src -O3 -fomit-frame-pointer -fstrict-aliasing -ffast-math -march=core2 -Wall -fexceptions -MT src/types.lo -MD -MP -MF src/.deps/types.Tpo -c ../src/types.c  -fPIC -DPIC -o src/.libs/types.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I. -I../include -Iinclude -I../src -O3 -fomit-frame-pointer -fstrict-aliasing -ffast-math -march=core2 -Wall -fexceptions -MT src/x86/ffi64.lo -MD -MP -MF src/x86/.deps/ffi64.Tpo -c ../src/x86/ffi64.c  -fPIC -DPIC -o src/x86/.libs/ffi64.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I. -I../include -Iinclude -I../src -O3 -fomit-frame-pointer -fstrict-aliasing -ffast-math -march=core2 -Wall -fexceptions -MT src/prep_cif.lo -MD -MP -MF src/.deps/prep_cif.Tpo -c ../src/prep_cif.c  -fPIC -DPIC -o src/.libs/prep_cif.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I. -I../include -Iinclude -I../src -O3 -fomit-frame-pointer -fstrict-aliasing -ffast-math -march=core2 -Wall -fexceptions -MT src/raw_api.lo -MD -MP -MF src/.deps/raw_api.Tpo -c ../src/raw_api.c  -fPIC -DPIC -o src/.libs/raw_api.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I. -I../include -Iinclude -I../src -O3 -fomit-frame-pointer -fstrict-aliasing -ffast-math -march=core2 -Wall -fexceptions -MT src/closures.lo -MD -MP -MF src/.deps/closures.Tpo -c ../src/closures.c  -fPIC -DPIC -o src/.libs/closures.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I. -I../include -Iinclude -I../src -I. -I../include -Iinclude -I../src -MT src/x86/unix64.lo -MD -MP -MF src/x86/.deps/unix64.Tpo -c ../src/x86/unix64.S  -fPIC -DPIC -o src/x86/.libs/unix64.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I. -I../include -Iinclude -I../src -I. -I../include -Iinclude -I../src -MT src/x86/unix64.lo -MD -MP -MF src/x86/.deps/unix64.Tpo -c ../src/x86/unix64.S -o src/x86/unix64.o >/dev/null 2>&1
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I. -I../include -Iinclude -I../src -O3 -fomit-frame-pointer -fstrict-aliasing -ffast-math -march=core2 -Wall -fexceptions -MT src/types.lo -MD -MP -MF src/.deps/types.Tpo -c ../src/types.c -o src/types.o >/dev/null 2>&1
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I. -I../include -Iinclude -I../src -O3 -fomit-frame-pointer -fstrict-aliasing -ffast-math -march=core2 -Wall -fexceptions -MT src/raw_api.lo -MD -MP -MF src/.deps/raw_api.Tpo -c ../src/raw_api.c -o src/raw_api.o >/dev/null 2>&1
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I. -I../include -Iinclude -I../src -O3 -fomit-frame-pointer -fstrict-aliasing -ffast-math -march=core2 -Wall -fexceptions -MT src/java_raw_api.lo -MD -MP -MF src/.deps/java_raw_api.Tpo -c ../src/java_raw_api.c -o src/java_raw_api.o >/dev/null 2>&1
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I. -I../include -Iinclude -I../src -O3 -fomit-frame-pointer -fstrict-aliasing -ffast-math -march=core2 -Wall -fexceptions -MT src/prep_cif.lo -MD -MP -MF src/.deps/prep_cif.Tpo -c ../src/prep_cif.c -o src/prep_cif.o >/dev/null 2>&1
../src/closures.c: In function 'dlmmap_locked':
../src/closures.c:488:7: warning: ignoring return value of 'ftruncate' declared with attribute 'warn_unused_result' [-Wunused-result]
  488 |       ftruncate (execfd, offset);
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~
../src/closures.c:500:7: warning: ignoring return value of 'ftruncate' declared with attribute 'warn_unused_result' [-Wunused-result]
  500 |       ftruncate (execfd, offset);
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I. -I../include -Iinclude -I../src -O3 -fomit-frame-pointer -fstrict-aliasing -ffast-math -march=core2 -Wall -fexceptions -MT src/x86/ffi64.lo -MD -MP -MF src/x86/.deps/ffi64.Tpo -c ../src/x86/ffi64.c -o src/x86/ffi64.o >/dev/null 2>&1
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I. -I../include -Iinclude -I../src -O3 -fomit-frame-pointer -fstrict-aliasing -ffast-math -march=core2 -Wall -fexceptions -MT src/closures.lo -MD -MP -MF src/.deps/closures.Tpo -c ../src/closures.c -o src/closures.o >/dev/null 2>&1
/bin/sh ./libtool  --tag=CC   --mode=link gcc  -O3 -fomit-frame-pointer -fstrict-aliasing -ffast-math -march=core2  -Wall -fexceptions   -o libffi_convenience.la  src/prep_cif.lo src/types.lo src/raw_api.lo src/java_raw_api.lo src/closures.lo                                src/x86/ffi64.lo src/x86/unix64.lo src/x86/ffi.lo src/x86/sysv.lo          
/bin/sh ./libtool  --tag=CC   --mode=link gcc  -O3 -fomit-frame-pointer -fstrict-aliasing -ffast-math -march=core2  -Wall -fexceptions -no-undefined -version-info `grep -v '^#' ../libtool-version`    -o libffi.la -rpath /home/techno/code/joinmarket-clientserver/jmvenv/lib/../lib64 src/prep_cif.lo src/types.lo src/raw_api.lo src/java_raw_api.lo src/closures.lo                                src/x86/ffi64.lo src/x86/unix64.lo src/x86/ffi.lo src/x86/sysv.lo          
libtool: link: rm -fr  .libs/libffi_convenience.a .libs/libffi_convenience.la
libtool: link: rm -fr  .libs/libffi.a .libs/libffi.la .libs/libffi.lai .libs/libffi.so .libs/libffi.so.6 .libs/libffi.so.6.0.4
libtool: link: ar cr .libs/libffi_convenience.a src/.libs/prep_cif.o src/.libs/types.o src/.libs/raw_api.o src/.libs/java_raw_api.o src/.libs/closures.o src/x86/.libs/ffi64.o src/x86/.libs/unix64.o src/x86/.libs/ffi.o src/x86/.libs/sysv.o 
libtool: link: gcc -shared  -fPIC -DPIC  src/.libs/prep_cif.o src/.libs/types.o src/.libs/raw_api.o src/.libs/java_raw_api.o src/.libs/closures.o src/x86/.libs/ffi64.o src/x86/.libs/unix64.o src/x86/.libs/ffi.o src/x86/.libs/sysv.o    -O3 -march=core2   -Wl,-soname -Wl,libffi.so.6 -o .libs/libffi.so.6.0.4
libtool: link: ranlib .libs/libffi_convenience.a
libtool: link: (cd ".libs" && rm -f "libffi.so.6" && ln -s "libffi.so.6.0.4" "libffi.so.6")
libtool: link: (cd ".libs" && rm -f "libffi.so" && ln -s "libffi.so.6.0.4" "libffi.so")
libtool: link: ar cr .libs/libffi.a  src/prep_cif.o src/types.o src/raw_api.o src/java_raw_api.o src/closures.o src/x86/ffi64.o src/x86/unix64.o src/x86/ffi.o src/x86/sysv.o
libtool: link: ( cd ".libs" && rm -f "libffi_convenience.la" && ln -s "../libffi_convenience.la" "libffi_convenience.la" )
libtool: link: ranlib .libs/libffi.a
libtool: link: ( cd ".libs" && rm -f "libffi.la" && ln -s "../libffi.la" "libffi.la" )
gmake[3]: Leaving directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu'
gmake[2]: Leaving directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu'
gmake[1]: Leaving directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu'
MAKE x86_64-pc-linux-gnu : 0 * check
gmake[1]: Entering directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu'
Making check in include
gmake[2]: Entering directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu/include'
gmake[2]: Nothing to be done for 'check'.
gmake[2]: Leaving directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu/include'
Making check in testsuite
gmake[2]: Entering directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu/testsuite'
gmake  check-DEJAGNU
gmake[3]: Entering directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu/testsuite'
Making a new site.exp file ...
srcdir='../../testsuite'; export srcdir; \
EXPECT=expect; export EXPECT; \
if /bin/sh -c "runtest --version" > /dev/null 2>&1; then \
  exit_status=0; l='libffi'; for tool in $l; do \
    if runtest --tool $tool --srcdir $srcdir  ; \
    then :; else exit_status=1; fi; \
  done; \
else echo "WARNING: could not find 'runtest'" 1>&2; :;\vim 
fi; \
exit $exit_status
WARNING: Couldn't find the global config file.
Using ../../testsuite/lib/libffi.exp as tool init file.
Test run by techno on Fri Jan  3 22:28:40 2025
Native configuration is x86_64-pc-linux-gnu

                === libffi tests ===

Schedule of variations:
    unix

Running target unix
Using /usr/share/dejagnu/baseboards/unix.exp as board description file for target.
Using /usr/share/dejagnu/config/unix.exp as generic interface file for target.
Using ../../testsuite/config/default.exp as tool-and-target-specific interface file.
Running ../../testsuite/libffi.call/call.exp ...
FAIL: libffi.call/cls_many_mixed_args.c -W -Wall -Wno-psabi -O0 (test for excess errors)
FAIL: libffi.call/cls_many_mixed_args.c -W -Wall -Wno-psabi -O2 (test for excess errors)
FAIL: libffi.call/cls_many_mixed_args.c -W -Wall -Wno-psabi -O3 (test for excess errors)
FAIL: libffi.call/cls_many_mixed_args.c -W -Wall -Wno-psabi -Os (test for excess errors)
FAIL: libffi.call/cls_many_mixed_args.c -W -Wall -Wno-psabi -O2 -fomit-frame-pointer (test for excess errors)

                === libffi Summary ===

# of expected passes            1905
# of unexpected failures        5
# of unsupported tests          28
gmake[3]: *** [Makefile:412: check-DEJAGNU] Error 1
gmake[3]: Leaving directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu/testsuite'
gmake[2]: *** [Makefile:488: check-am] Error 2
gmake[2]: Leaving directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu/testsuite'
gmake[1]: *** [Makefile:1590: check-recursive] Error 1
gmake[1]: Leaving directory '/home/techno/code/joinmarket-clientserver/deps/libffi-3.2.1/x86_64-pc-linux-gnu'
gmake: *** [Makefile:3109: check] Error 2
Libffi was not built. Exiting.

```

bumped libffi to latest and it builds just fine

    libffi_version='libffi-3.4.6'
    libffi_lib_tar="v3.4.6.tar.gz"
    libffi_lib_sha='9ac790464c1eb2f5ab5809e978a1683e9393131aede72d1b0a0703771d3c6cda'

--- a/install.sh
+++ b/install.sh
@@ -228,42 +228,17 @@ dep_get ()
     popd || return 1
 }
 
-# add '--disable-docs' to libffi ./configure so makeinfo isn't needed
-# https://github.com/libffi/libffi/pull/190/commits/fa7a257113e2cfc963a0be9dca5d7b4c73999dcc
-libffi_patch_disable_docs ()
+libffi_autoreconf_patch ()
 {
-    cat <<'EOF' > Makefile.am.patch
-56c56,59
-< info_TEXINFOS = doc/libffi.texi
----
-> info_TEXINFOS =
-> if BUILD_DOCS
-> #info_TEXINFOS += doc/libffi.texi
-> endif
-EOF
-
     # autogen.sh is not happy when run from some directories, causing it
-    # to create an ltmain.sh file in our ${jm_root} directory.  weird.
+    # to create an ltmain.sh file in our ${jm_root} directory. weird.
     # https://github.com/meetecho/janus-gateway/issues/290#issuecomment-125160739
     # https://github.com/meetecho/janus-gateway/commit/ac38cfdae7185f9061569b14809af4d4052da700
     cat <<'EOF' > autoreconf.patch
 18a19
 > AC_CONFIG_AUX_DIR([.])
 EOF
-
-    cat <<'EOF' > configure.ac.patch
-545a546,552
-> AC_ARG_ENABLE(docs,
->               AC_HELP_STRING([--disable-docs],
->                              [Disable building of docs (default: no)]),
->               [enable_docs=no],
->               [enable_docs=yes])
-> AM_CONDITIONAL(BUILD_DOCS, [test x$enable_docs = xyes])
-> 
-EOF
-    patch Makefile.am Makefile.am.patch
     patch configure.ac autoreconf.patch
-    patch configure.ac configure.ac.patch
 }
 
 libffi_build ()
@@ -279,9 +254,9 @@ libffi_build ()
 
 libffi_install ()
 {
-    libffi_version='libffi-3.2.1'
-    libffi_lib_tar="v3.2.1.tar.gz"
-    libffi_lib_sha='96d08dee6f262beea1a18ac9a3801f64018dc4521895e9198d029d6850febe23'
+    libffi_version='libffi-3.4.6'
+    libffi_lib_tar="v3.4.6.tar.gz"
+    libffi_lib_sha='9ac790464c1eb2f5ab5809e978a1683e9393131aede72d1b0a0703771d3c6cda'
     libffi_url="https://github.com/libffi/libffi/archive"
 
     if check_skip_build "${libffi_version}"; then
@@ -291,7 +266,7 @@ libffi_install ()
         return 1
     fi
     pushd "${libffi_version}" || return 1
-    if ! libffi_patch_disable_docs; then
+    if ! libffi_autoreconf_patch; then
         return 1
     fi
     if libffi_build; then

